### PR TITLE
[SPARK-43591][SQL] Assign a name to the error class _LEGACY_ERROR_TEMP_0013

### PR DESCRIPTION
**Pull request by panbingkun**
### What changes were proposed in this pull request?
The pr aims to assign a name to the error class _LEGACY_ERROR_TEMP_0013.

### Why are the changes needed?
The changes improve the error framework.

### Does this PR introduce _any_ user-facing change?
No.

### How was this patch tested?
Pass GA.

### CODE REVIEW

File: core/src/main/resources/error/error-classes.json [hunk 1]
```diff
@@ -5627,4 +5627,4 @@
       "Failed to get block <blockId>, which is not a shuffle block"
     ]
   }
-}
+}
```
Comment (MaxGekk) [inline core/src/main/resources/error/error-classes.json:5630]: Unnecessary change. Please, revert it back.

**Review by MaxGekk — APPROVED** [review]
LGTM except of a minor comment.

File: core/src/main/resources/error/error-classes.json (hunk 1 above)
Comment (panbingkun) [reply core/src/main/resources/error/error-classes.json]: This is done.

File: core/src/main/resources/error/error-classes.json [hunk 2]
```diff
@@ -1877,6 +1877,11 @@
           "JOIN USING with LATERAL correlation."
         ]
       },
+      "LATERAL_WITH_PIVOT" : {
```
Comment (MaxGekk) [inline core/src/main/resources/error/error-classes.json:1880]: The main point is the combination is not allowed in `FROM`. Since there are 3 related errors:
- unpivotWithPivotInFromClauseNotAllowedError
- lateralWithPivotInFromClauseNotAllowedError
- lateralWithUnpivotInFromClauseNotAllowedError

Could you add the error class `NOT_ALLOWED_IN_FROM`, and add sub-classes to it, please.

**Review by MaxGekk — CHANGES_REQUESTED** [review]

File: core/src/main/resources/error/error-classes.json (hunk 2 above)
Comment (panbingkun) [reply core/src/main/resources/error/error-classes.json:1880]: Ok, Let me do it

**Review by MaxGekk — CHANGES_REQUESTED** [review]

File: core/src/main/resources/error/error-classes.json (hunk 1 above)
Comment (panbingkun) [reply core/src/main/resources/error/error-classes.json]: I have proposed a new PR to address this issue
PR: https://github.com/apache/spark/pull/41256

**Review by MaxGekk — APPROVED** [review]
Waiting for CI.

**Comment by MaxGekk** [comment]
+1, LGTM. Merging to master.
Thank you, @panbingkun.
